save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/PluginConfig.kt

```diff
@@ -4,27 +4,25 @@
 
 package org.cqfn.save.core.plugin
 
+import org.cqfn.save.core.config.TestConfigSections
+
 import kotlinx.serialization.Serializable
 
 /**
- * Core interface for plugin configuration
+ * Core interface for plugin configuration (like warnPlugin/fixPLuin/e.t.c)
  */
-interface PluginConfig<T : PluginConfig<T>> {
+@Suppress("INLINE_CLASS_CAN_BE_USED")
+interface PluginConfig {
     /**
-     * Merge current config into [childConfig]
-     *
-     * @param childConfig list of child configs, which will be filtered for merging
+     * type of the config (usually related to the class: WARN/FIX/e.t.c)
      */
-    @Suppress("TYPE_ALIAS")
-    fun mergeConfigInto(childConfig: MutableList<PluginConfig<*>>)
+    val type: TestConfigSections
```

Comment:
Won't it cause troubles, when we start implementing custom plugins injection? Maybe, instead of enum, we can have a marker interface here, that still could be used to compare plugins, but will be extendible?

Reply:
yes, very good point I see the problem, but let's leave it as is by for now... 
Anyway it will be a hevay change